gcp_airflow_foundations/base_class/source_template_config.py

```diff
@@ -0,0 +1,152 @@
+from airflow.exceptions import AirflowException
+
+from dacite import Config
+from dataclasses import dataclass, field
+from dacite import from_dict
+
+
+from pydantic import validator, root_validator
+
+import datetime
+import re
+from typing import List, Optional
+
+from gcp_airflow_foundations.base_class.ods_metadata_config import (
+    OdsTableMetadataConfig,
+)
+from gcp_airflow_foundations.enums.ingestion_type import IngestionType
+from gcp_airflow_foundations.enums.hds_table_type import HdsTableType
+
+from gcp_airflow_foundations.base_class.ods_table_config import OdsTableConfig
+from gcp_airflow_foundations.base_class.hds_table_config import HdsTableConfig
+from gcp_airflow_foundations.base_class.facebook_table_config import FacebookTableConfig
+from gcp_airflow_foundations.base_class.template_ingestion_options_config import TemplateIngestionOptionsConfig
+from gcp_airflow_foundations.base_class.column_udf_config import ColumnUDFConfig
+
+
+@dataclass
+class SourceTemplateConfig:
+    """
+    Template configuration data class.
+
+    Attributes:
+        table_names : List of table name. Used for Dag Id/s.
+        dest_table_override: Optional target table name. If None, use table_name instead.
+        surrogate_keys : Keys used to identify unique records when merging into ODS.
+        column_mapping : Mapping used to rename columns.
+        cluster_fields: The fields used for clustering. BigQuery supports clustering for both partitioned and non-partitioned tables.
+        landing_zone_table_name_override_template: Optional staging zone table name template.
+        column_casting : Mapping used to cast columns into a specific data type. Note column name uses that of the landing zone table.
+        ods_config : ODS table configuration. See :class:`gcp_airflow_foundations.base_class.ods_table_config.OdsTableConfig`.
+        hds_config : HDS table configuration. See :class:`gcp_airflow_foundations.base_class.hds_table_config.HdsTableConfig`.
+        template_ingestion_options: Configuration for template-level ingestion.
+        facebook_table_config: Extra options for ingesting data from the Facebook API.
```

Comment:
what's this facebook stuff?

Reply:
@danielkulikov what happened  to removing this?

Reply:
what happened to removing facebook_table_config?

Reply:
removed it from SourceTemplateConfig now, but kept it and gave it a default arg in SourceTableConfig. Don't wanna break old setup, so can address that in a secondary PR